### ISSUE TITLE
trailing-comma adds comma to when condition

### ISSUE THREAD

**Issue by dalewking** [pinterest/ktlint#1312]
Not sure whether to consider this a bug or not, but I found it unusual, so thought I should report it. So I enabled the trailing-comma rule and let ktlint autocorrect the code. So there was some code like this:
```
    when(foo) {
        SOME_VALUE1,
        SOME_VALUE2,
        SOME_VALUE3 -> { ... }
    }
```

After running ktlint auto-correct it changed the code to this:
```
    when(foo) {
        SOME_VALUE1,
        SOME_VALUE2,
        SOME_VALUE3, -> { ... }
    }
```

That trailing comma before the `->` really serves no purpose. The purpose of the trailing comma is to reduce the number of lines changed when adding a new value, but that will not happen in this case since the `->` will have to move

So the trailing-comma rule should probably ignore a line if the comma would not be at the end of the line.

**Comment by paul-dingemans** [comment]
The code currently generated by ktlint is not accepted by the default IntelliJ formatting which changes it to
```
    when (foo) {
        SOME_VALUE1,
        SOME_VALUE2,
        SOME_VALUE3,
        -> {
            ...
        }
    }
```
With this format it is possible to add an element on a new line anywhere in the list of value without affecting any other lines.

Ktlint should generate this format in case the trailing comma setting is enabled.
